providers/dns/route53/route53.go

```diff
@@ -104,16 +104,19 @@ func (r *DNSProvider) getHostedZoneID(fqdn string) (string, error) {
 		zones = append(zones, zoneResp.HostedZones...)
 	}
 
+	authZone, err := acme.FindZoneByFqdn(fqdn, acme.RecursiveNameserver)
+	if err != nil {
+		return "", err
+	}
+
 	var hostedZone route53.HostedZone
 	for _, zone := range zones {
-		if strings.HasSuffix(fqdn, zone.Name) {
-			if len(zone.Name) > len(hostedZone.Name) {
-				hostedZone = zone
-			}
+		if zone.Name == authZone {
```

Comment:
Why aren't we querying route53 with the hosted zone we determined before we get here? From what I gathered from a quick look at the docs it seems like we can pass a marker to the `ListHostedZones` function.